### PR TITLE
Add `thumbv6m-none-eabi` experimental target

### PR DESCRIPTION
**Pull request by Hoverbear**
We discussed adding this in #1033

### CODE REVIEW

File: ferrocene/ci/scripts/calculate_parameters.py [hunk 1]
```diff
@@ -44,6 +44,7 @@
 
 GENERIC_BUILD_STD_TARGETS = [
     "aarch64-unknown-none",
+    "thumbv6m-none-eabi",
     "thumbv7em-none-eabi",
```
Comment (jonathanpallant) [inline ferrocene/ci/scripts/calculate_parameters.py:48]: Maybe we could sort this array? Just so it's obvious where new targets need to go.

File: ferrocene/doc/release-notes/src/next.rst [hunk 2]
```diff
@@ -27,4 +27,5 @@ shipped as a preview.
   * :target-with-triple:`thumbv8m.base-none-eabi`
   * :target-with-triple:`thumbv8m.main-none-eabi`
   * :target-with-triple:`thumbv8m.main-none-eabihf`
+  * :target-with-triple:`thumbv6m-none-eabi`
```
Comment (jonathanpallant) [inline ferrocene/doc/release-notes/src/next.rst:30]: 6 comes before 8?

File: ferrocene/doc/target-names.toml [hunk 3]
```diff
@@ -14,6 +14,7 @@ armv8r-none-eabihf          = "Armv8-R bare-metal (hard-float)"
 "thumbv8m.base-none-eabi"   = "Armv8-M Baseline bare-metal"
```
Comment (jonathanpallant) [inline ferrocene/doc/target-names.toml:14]: Maybe we should sort this one too (ignoring the `"`)?

File: ferrocene/doc/user-manual/src/targets/index.rst [hunk 4]
```diff
@@ -167,6 +155,24 @@ should not be used in production.
      - Bare-metal
      - \-
 
+   * - :target:`thumbv7em-none-eabi`
```
Comment (jonathanpallant) [inline ferrocene/doc/user-manual/src/targets/index.rst:158]: What's the intended ordering for this table?

File: ferrocene/packages.toml [hunk 5]
```diff
@@ -69,6 +69,7 @@ subset = "default"
 targets = [
   "aarch64-unknown-none",
   "aarch64-unknown-linux-gnu",
+  "thumbv6m-none-eabi",
```
Comment (jonathanpallant) [inline ferrocene/packages.toml:72]: Hello, me again, asking about ordering.

File: ferrocene/tools/self-test/src/targets.rs [hunk 6]
```diff
@@ -33,6 +33,7 @@ static SUPPORTED_TARGETS: &[TargetSpec] = &[
     TargetSpec { triple: "thumbv8m.main-none-eabihf", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "thumbv7em-none-eabihf", std: false, linker: Linker::BundledLld },
     TargetSpec { triple: "thumbv7em-none-eabi", std: false, linker: Linker::BundledLld },
+    TargetSpec { triple: "thumbv6m-none-eabi", std: false, linker: Linker::BundledLld },
```
Comment (jonathanpallant) [inline ferrocene/tools/self-test/src/targets.rs:36]: Guess what.

**Comment by jonathanpallant** [comment]
I have no problem with the notes you played, only the order you played them in.

... except what about the list at https://github.com/ferrocene/ferrocene/blob/7362cba13cde4b0cb8afdc0093f493ffaeb88d61/ferrocene/ci/configure.sh#L156?

**Comment by Hoverbear** [comment]
Oh, great catch. (Also, appreciate the ordering prompt.)

File: ferrocene/doc/user-manual/src/targets/index.rst [hunk 7]
```diff
@@ -119,38 +118,39 @@ should not be used in production.
...
-     - ``thumbv8m.base-none-eabi``
+   * - :target:`armv7r-none-eabihf`
+     - ``armv7r-none-eabihf``
      - Cross-compilation
      - Bare-metal
      - \-
 
-   * - :target:`thumbv8m.main-none-eabi`
-     - ``thumbv8m.main-none-eabi``
+   * - :target:`armebv7r-none-eabihf`
```
Comment (jonathanpallant) [inline ferrocene/doc/user-manual/src/targets/index.rst:133]: See note above.

File: ferrocene/doc/target-names.toml [hunk 8]
```diff
@@ -7,22 +7,23 @@
 # In the documents, as long as the ferrocene_qualification extension is added,
 # you can refer to them with :target:`triple`.
 
+
+aarch64-apple-darwin        = "Apple Silicon macOS"
+aarch64-unknown-linux-gnu   = "Armv8-A Linux (glibc)"
 aarch64-unknown-none        = "Armv8-A bare-metal"
+aarch64-unknown-nto-qnx710  = "Armv8-A QNX® Neutrino® 7.1.0"
+armebv7r-none-eabihf        = "Armv7-R bare-metal (hard-float, big-endian)"
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply ferrocene/doc/target-names.toml:15]: Ooops, this was my fault for just using "sort" here but not in the manual.

**Comment by jonathanpallant** [comment]
Whilst checking that the re-ordering didn't drop anything from the self-test's list of targets, I saw that `TargetSpec { triple: "x86_64-unknown-linux-gnu", std: true, linker: Linker::HostCc },` was still at the top of the list. Maybe it should be down with the other instance of `x86_64-unknown-linux-gnu`. It looks like there are two versions of each Linux target - one for native compilation and one for cross compilation.

**Review by jonathanpallant — APPROVED** [review]

**Comment by Hoverbear** [comment]
bors merge

**Pull request merged**